flink-runtime/src/main/java/org/apache/flink/runtime/security/token/KerberosDelegationTokenManager.java

```diff
@@ -110,13 +126,84 @@ public void obtainDelegationTokens(Credentials credentials) {
      * task managers.
      */
     @Override
-    public void start() {
-        LOG.info("Starting renewal task");
+    public void start() throws Exception {
+        checkState(renewalExecutor == null, "Manager is already started");
+
+        if (!isRenewalPossible()) {
+            LOG.info("Renewal is NOT possible, skipping to start renewal task");
+            return;
+        }
+
+        ThreadFactory threadFactory =
```

Comment:
nit: Please mark variables as `final` where possible.

Reply:
Since we plan to use existing executors this code will be deleted.